### PR TITLE
Fixes #14072 - Use to_hash when requesting session original_uri

### PR DESCRIPTION
**Pull request by shlomizadok**

### CODE REVIEW

**Comment by dLobatog** [comment]
@shlomizadok Nice, this fixes the problem :+1: Can you add a test for this? It broke during the Rails 4 upgrade precisely because there was no test for it. I suppose you can stub `session_expiry` in the test.

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -138,7 +138,7 @@ def find_resource(permission = :view_users)
 
   def login_user(user)
     session[:user]         = user.id
-    uri                    = session[:original_uri]
+    uri                    = session.to_hash[:original_uri] || session.to_hash['original_uri']
```
Comment (shlomizadok) [inline app/controllers/users_controller.rb]: https://garrickvanburen.com/passing-sessions-and-referers-in-rails-functional-tests/
Old, but still true: In test, session keys are strings

Comment (tbrisker) [inline app/controllers/users_controller.rb]: why not `to_hash.with_indifferent_access[:original_uri]`?

Comment (shlomizadok) [inline app/controllers/users_controller.rb]: Because I forgot about it?
Changing

**Comment by shlomizadok** [comment]
@tbrisker - thanks. Changed

**Comment by tbrisker** [comment]
Merged as fb194fd649683b95c474bc903798d166fd5fa430, thanks @shlomizadok!

**Pull request closed without merging**